src/typecheck/core.ml

```diff
@@ -12,12 +12,17 @@ module Ae = struct
       (Ty : Dolmen.Intf.Ty.Ae_Base with type t = Type.Ty.t)
       (T : Dolmen.Intf.Term.Ae_Base with type t = Type.T.t) = struct
 
+    let mk_or a b = T._or [a; b]
+    let mk_and a b = T._and [a; b]
+
     let parse env s =
       match s with
 
       (* Types *)
       | Type.Builtin Ast.Bool ->
         `Ty (Base.app0 (module Type) env s Ty.bool)
+      | Type.Builtin Ast.Prop ->
+        `Ty (Base.app0 (module Type) env s Ty.bool)
```

Comment:
Not that it's too big of a problem, but are there cases where we would typecheck something that alt-ergo wouldn't because here we don't make a distinction between bool and prop ? (In any case, it's probably better to leave this as is, and later on be careful when translating typed terms to alt-ergo's hashconsed terms).

Reply:
I don't think so. According to [this](https://github.com/OCamlPro/alt-ergo/blob/next/docs/sphinx_docs/Input_file_formats/Native/01_declaration_of_symbols.md#predicate-propositional-valued-functions) prop and bool used to be two different types but are now the same and behave in the same way. And internally in Alt-Ergo there is only a bool type and no prop while it still supports prop in it's input files.

Reply:
Ok, great